src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/BaseRecalibratorDataflow2.java

```diff
@@ -0,0 +1,282 @@
+                .apply(ParDo.of(new DoFnWLog<BaseRecalOutput, RecalibrationTables>("getTables") {
+                    private static final long serialVersionUID = 1L;
+
+                    @Override
+                    public void processElement(ProcessContext c) {
+                        final BaseRecalOutput br = c.element();
+                        c.output(br.getRecalibrationTables());
+                    }
+                }));
+
+            // If saving textual output then we need to make sure we can get to the output
+            if (saveTextualTables) {
+                if (null == outputTablesPath) {
+                    // we need those, so let's pick a location for them.
+                    outputTablesPath = pickOutputTablesPath(isRemote(), stagingLocation);
+                }
+                DataflowUtils.SaveDestination dest = DataflowUtils.serializeSingleObject(recalibrationTable, outputTablesPath);
+                if (isRemote() && dest == DataflowUtils.SaveDestination.LOCAL_DISK) {
+                    throw new UserException("If running on the cloud, either leave outputTablesPath unset or point it to a GCS location.");
+                }
+            }
+            bunny.stepEnd("setup");
+        } catch (UserException rx) {
+            throw rx;
+        } catch (GATKException rx) {
+            throw rx;
+        } catch (Exception x) {
+            throw new GATKException("Unexpected: " + x.getMessage(), x);
+        }
+    }
+
+    @Override
+    protected void afterPipeline(Pipeline p) {
+        bunny.stepEnd("dataflow");
+        if (saveTextualTables) {
+            logger.info("Saving recalibration report to "+BRAC.RAC.RECAL_TABLE_FILE.getName());
+            //  Get the table back and output it in text form to RAC.RECAL_TABLE.
+            // TODO: if running on the cloud and the output destination is on the cloud, then it's faster to have a worker do it directly, without the file roundtrip.
+            try (ObjectInputStream oin = new ObjectInputStream(BucketUtils.openFile(outputTablesPath, p.getOptions()))) {
+                Object o = oin.readObject();
+                RecalibrationTables rt = (RecalibrationTables) o;
+                BaseRecalibratorFn.SaveTextualReport(BRAC.RAC.RECAL_TABLE_FILE, readsHeader, rt,  BRAC);
+                bunny.stepEnd("repatriate_report");
+            } catch (Exception e) {
+                throw new GATKException("Unexpected: unable to read results file. (bug?)", e);
+            }
+        } else {
+            // because the user didn't ask us to.
+            logger.info("Not saving recalibration report");
+        }
+        bunny.end();
+    }
+
+    protected static String pickOutputTablesPath(boolean remote, String stagingLocation) {
+        if (remote) {
+            return GcsPath.fromUri(stagingLocation).resolve(TEMP_RECALTABLES + ".sj").toString();
+        } else {
+            File outputTables = IOUtils.createTempFile(TEMP_RECALTABLES, ".sj");
```

Comment:
Yes. This is called when the user didn't ask us to save this intermediate result but we happen to need it. I've renamed the method to make this clearer.